compiler/rustc_typeck/src/check/compare_method.rs

```diff
@@ -225,7 +225,7 @@ fn compare_predicate_entailment<'tcx>(
         let (impl_m_own_bounds, _) = infcx.replace_bound_vars_with_fresh_vars(
             impl_m_span,
             infer::HigherRankedType,
-            ty::Binder::bind(impl_m_own_bounds.predicates, tcx),
+            ty::Binder::dummy(impl_m_own_bounds.predicates),
         );
```

Comment:
This is weird. If we don't ever have bound vars, why do we have to `replace_bound_vars_with_fresh_vars`

Reply:
Okay, I think we can remove `replace_bound_vars_with_fresh_vars` here because `impl_m_predicates` is simply the `Predicates` of the impl, which obviously won't have escaping bound vars. And `impl_to_placeholder_substs` will only have substs for the impl itself, none of which are late-bound.